### ISSUE TITLE
App crashes during animations after pull sync

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#1784]
- [ ] scenario 1
    - create two TEs: X and Y on some date (doesn't matter if you create them on the webapp or in the mobile app)
    - sync the mobile app
    - change the date of both those TEs to a different date in the web app
    - sync the mobile app
    - :boom:
- [ ] scenario 2
    - create two TEs: U and V on 27. 3. (assuming you already don't have no other TEs on that date, otherwise pick some other suitable date, as in the previous case it doesn't matter if it is in the webapp or    the mobile app)
    - sync the mobile app
    - change the date of U to 28. 3. in the web app (also must be emtpy before this step)
    - change the date of V to 29. 3. in the web app (also must be empty before this step)
    - sync the mobile app
    - :boom:
- [ ] scenario 3
    - create two TEs: M and N on two different dates where no other TEs are yet
    - sync the mobile app
    - delete M and N in the web app
    - sync the mobile app
    - :boom:

**Comment by willsb** [comment]
As discussed on Slack, this is being put on hold for the time being

**Comment by amulware** [comment]
Some potentially useful data:

**Exception and stack trace with animations enabled**
```
UITableView.g.cs - line 419
NSInternalInconsistencyException: Invalid update: invalid number of rows in section 0. The number of rows contained in an existing section after the update (2) must be equal to the number of rows contained in that section before the update (2), plus or minus the number of rows inserted or deleted from that section (1 inserted, 0 deleted) and plus or minus the number of rows moved into or out of that section (0 moved in, 0 moved out).
LAST EXCEPTION BACKTRACE
CoreFoundation
__exceptionPreprocess
libobjc.A.dylib
objc_exception_throw
CoreFoundation
+[NSException raise:format:arguments:]
Foundation
-[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
UIKit
-[UITableView _endCellAnimationsWithContext:]
UIKit
-[UITableView _updateRowsAtIndexPaths:withUpdateAction:rowAnimation:usingPresentationValues:]
Toggl.Daneel
UITableView.g.cs:419
Toggl.Daneel
Toggl_Daneel_ViewSources_GroupedCollectionTableViewSource_2__c__DisplayClass22_0_TCollection_REF_TItem_REF__OnChildCollectionChangedb__0 <unknown>:1
...
```


**Exception and stack trace with animations disabled**
```
TimeEntriesLogViewSource.cellIdentifierForItem (System.Object item)
System.ArgumentException: Unexpected item type. Must be of type TimeEntryViewModel
TimeEntriesLogViewSource.cellIdentifierForItem (System.Object item)
TimeEntriesLogViewSource.GetOrCreateCellFor (UIKit.UITableView tableView, Foundation.NSIndexPath indexPath, System.Object item)
TimeEntriesLogViewSource.GetCell (UIKit.UITableView tableView, Foundation.NSIndexPath indexPath)
MainTableViewSource.GetCell (UIKit.UITableView tableView, Foundation.NSIndexPath indexPath)
(wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate)
UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName)
Application.Main (System.String[] args)
```

Relevant files:

https://github.com/toggl/mobileapp/blob/9103892a8701890b1655979a3eb3492191142a3c/Toggl.Daneel/ViewSources/TimeEntriesLogViewSource.cs

https://github.com/toggl/mobileapp/blob/9103892a8701890b1655979a3eb3492191142a3c/Toggl.Daneel/ViewSources/GroupedCollectionTableViewSource.cs

**Comment by simonrozsival** [comment]
The fix for this bug was shipped with 1.4.3 and according to AppCenter this does not cause crashes anymore.
